src/controllers/GovernanceTimelock.sol

```diff
@@ -0,0 +1,158 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity >=0.8.9;
+
+/// interfaces
+import "../interfaces/IGovernanceTimelock.sol";
+
+/// libraries
+import "../libraries/Error.sol";
+
+contract GovernanceTimelock is IGovernanceTimelock {
+    /*///////////////////////////////////////////////////////////////
+                             CONSTANTS
+    //////////////////////////////////////////////////////////////*/
+    uint256 public constant MINIMUM_DELAY = 2 days;
+    uint256 public constant MAXIMUM_DELAY = 30 days;
+
+    /// @notice The time window within which a transaction can be executed, following its ETA. Beyond this point a transaction is considered stale and cannot be executed.
+    uint256 public constant GRACE_PERIOD = 14 days;
+
+    /*/////////////////////////////////////////////////////////////////
+                            STATE VARIABLES
+    ////////////////////////////////////////////////////////////////*/
+    uint256 public txCounter;
```

Comment:
do we still need `txCounter`? we can use the hash of the transaction as the transaction ID now, right?